manifests/plugin/migration.pp

```diff
@@ -1,5 +1,47 @@
 # @summary Pulp 2 to Pulp 3 Migration Plugin
-class pulpcore::plugin::migration {
+#
+# @param mongo_db_name
+#   MongoDB database name for Pulp2. Used for migrating Pulp2 content
+#
+# @param mongo_db_seeds
+#   MongoDB seeds. Used for migrating Pulp2 content
+#
+# @param mongo_db_username
+#   MongoDB username for Pulp2
+#
+# @param mongo_db_password
+#   Password of MongoDB user
+#
+# @param mongo_db_replica_set
+#   MongoDB Replica Set
+#
+# @param mongo_db_ssl
+#   Use SSL for MongoDB connection
+#
+# @param mongo_db_ssl_keyfile
+#   SSL Keyfile for MongoDB
+#
+# @param mongo_db_ssl_certfile
+#   SSL Certfile for MongoDB
+#
+# @param mongo_db_verify_ssl
+#   Whether SSL verification is required for MongoDB
+#
+# @param mongo_db_ca_path
+#   CA bundle path to use for MongoDB SSL connection
+#
+class pulpcore::plugin::migration (
+  String $mongo_db_name = 'pulp_database',
+  String $mongo_db_seeds = 'localhost:27017',
+  Optional[String] $mongo_db_username = undef,
+  Optional[String] $mongo_db_password = undef,
+  Optional[String] $mongo_replica_set = undef,
+  String $mongo_db_ssl = 'False',
```

Comment:
We should be using real booleans and only translate them in the templates. For example here `Boolean $mongo_db_ssl = false`. Then in the template `<%= @mongo_db_ssl ? 'True' : 'False' %>`. The same goes for the ssl verify parameter.